### PR TITLE
feature addition: add mariadb vector store

### PR DESCRIPTION
**Pull request by rusher**
Here is the MariaDB database store implementation. 

not just another one : https://mariadb.org/projects/mariadb-vector/ :)

### CODE REVIEW

File: spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/vectorstore/mariadb/MariaDbStoreAutoConfigurationIT.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+		.withConfiguration(AutoConfigurations.of(MariaDbStoreAutoConfiguration.class,
+				JdbcTemplateAutoConfiguration.class, DataSourceAutoConfiguration.class))
+		.withUserConfiguration(Config.class)
+		.withPropertyValues("spring.ai.vectorstore.mariadb.distanceType=COSINE",
+				"spring.ai.vectorstore.mariadb.initialize-schema=true",
+				// JdbcTemplate configuration
+				String.format("spring.datasource.url=jdbc:mariadb://%s:%d/%s", mariadbContainer.getHost(),
+						mariadbContainer.getMappedPort(3306), mariadbContainer.getDatabaseName()),
```
Comment (eddumelendez) [inline spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/vectorstore/mariadb/MariaDbStoreAutoConfigurationIT.java:72]:
```suggestion
				String.format("spring.datasource.url=%s", mariadbContainer.getJdbcUrl()),
```

File: vector-stores/spring-ai-mariadb-store/src/test/java/org/springframework/ai/vectorstore/MariaDBStoreCustomNamesIT.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		.withPassword("mariadbpwd")
+		.withDatabaseName(schemaName);
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+		.withUserConfiguration(TestApplication.class)
+		.withPropertyValues("test.spring.ai.vectorstore.mariadb.distanceType=COSINE",
+
+				// JdbcTemplate configuration
+				String.format("app.datasource.url=jdbc:mariadb://%s:%d/%s", mariadbContainer.getHost(),
+						mariadbContainer.getMappedPort(3306), schemaName),
```
Comment (eddumelendez) [inline vector-stores/spring-ai-mariadb-store/src/test/java/org/springframework/ai/vectorstore/MariaDBStoreCustomNamesIT.java:63]: ditto

File: vector-stores/spring-ai-mariadb-store/src/test/java/org/springframework/ai/vectorstore/MariaDBStoreIT.java [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+		.withPassword("mariadbpwd")
+		.withDatabaseName(schemaName);
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+		.withUserConfiguration(TestApplication.class)
+		.withPropertyValues("test.spring.ai.vectorstore.mariadb.distanceType=COSINE",
+
+				// JdbcTemplate configuration
+				String.format("app.datasource.url=jdbc:mariadb://%s:%d/%s?maxQuerySizeToLog=50000", mariadbContainer.getHost(),
+						mariadbContainer.getMappedPort(3306), schemaName),
```
Comment (eddumelendez) [inline vector-stores/spring-ai-mariadb-store/src/test/java/org/springframework/ai/vectorstore/MariaDBStoreIT.java:81]: ditto

File: spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/vectorstore/mariadb/MariaDbStoreAutoConfigurationIT.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+	@Test
+	public void addAndSearch() {
+
+		this.contextRunner.run(context -> {
+
+			MariaDBVectorStore vectorStore = context.getBean(MariaDBVectorStore.class);
+			TestObservationRegistry observationRegistry = context.getBean(TestObservationRegistry.class);
+
+			assertThat(isFullyQualifiedTableExists(context, null, MariaDBVectorStore.DEFAULT_TABLE_NAME)).isTrue();
```
[the remark this answers is not in the document]
Comment (rusher) [reply spring-ai-spring-boot-autoconfigure/src/test/java/org/springframework/ai/autoconfigure/vectorstore/mariadb/MariaDbStoreAutoConfigurationIT.java:110]: right, corrected

**Comment by rusher** [comment]
Right, i have corrected the test. Must i merge the commit into one ?

**Comment by tzolov** [comment]
Thanks @rusher , I'll give it a try.  Please don't squash the commits. I will do it before the merge

**Comment by tzolov** [comment]
Thank you @rusher !
Cleared, rebased, squashed and merged at 0b00e6f4466fac1a5acca173f2bede905e7b56b6

**Pull request closed without merging**